File: exploratory_analysis.m
function exploratory_analysis(data, data_updated)

    disp("Running Exploratory Data Analysis...");

    %% ---------------- CORRELATION HEATMAP ----------------
    correlationMatrix = corr(table2array(data_updated), 'Rows', 'complete');

    figure;
    h = heatmap(data_updated.Properties.VariableNames, ...
                data_updated.Properties.VariableNames, ...
                correlationMatrix);

    h.Colormap = parula;
    h.Title = 'Correlation Heatmap';
    h.FontSize = 12;
    h.ColorLimits = [-1, 1];

    %% ---------------- SCATTER PLOTS ----------------
    features = data_updated.Properties.VariableNames;

    figure;

    for i = 1:length(features)

        featureName = features{i};

        subplot(ceil(length(features)/3), 3, i);

        scatter(data_updated.(featureName), data_updated.RH, '.');
        hold on;

        % Fit linear model
        lm = fitlm(data_updated.(featureName), data_updated.RH);
        plot(lm);

        title(['RH vs ', featureName]);
        xlabel(featureName);
        ylabel('RH');

        legend off;
        hold off;
    end

    sgtitle('Feature vs RH Relationships');

end

File: exploratory_analysis.py
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

def exploratory_analysis(data):

    print("=== Exploratory Data Analysis ===")

    # -------------------------------
    # 1. BASIC INFO
    # -------------------------------
    print("\nDataset Info:")
    print(data.info())

    print("\nStatistical Summary:")
    print(data.describe())

    # -------------------------------
    # 2. CORRELATION HEATMAP
    # -------------------------------
    plt.figure(figsize=(12, 8))

    corr = data.corr()

    sns.heatmap(
        corr,
        annot=False,
        cmap="viridis",
        vmin=-1,
        vmax=1
    )

    plt.title("Correlation Heatmap")
    plt.savefig("correlation_heatmap.png")
    plt.show()

    # -------------------------------
    # 3. SCATTER PLOTS (FEATURE vs RH)
    # -------------------------------
    features = data.columns.drop(['RH'])

    n_cols = 3
    n_rows = (len(features) // n_cols) + 1

    plt.figure(figsize=(15, 10))

    for i, feature in enumerate(features, 1):
        plt.subplot(n_rows, n_cols, i)

        plt.scatter(data[feature], data['RH'], s=10)

        plt.xlabel(feature)
        plt.ylabel("RH")
        plt.title(f"{feature} vs RH")

    plt.tight_layout()
    plt.suptitle("Feature vs RH Relationships", y=1.02)
    plt.savefig("scatter_plots.png")
    plt.show()
